app/models/mixins/has_monitoring_manager_mixin.rb

```diff
@@ -0,0 +1,20 @@
+module HasMonitoringManagerMixin
+  extend ActiveSupport::Concern
+
+  private
+
+  def ensure_managers
+    if try(:monitoring_manager_needed?)
+      build_monitoring_manager
+      monitoring_manager.name = "#{name} Monitoring Manager"
```

Comment:
if possible I'd like to keep this similar to https://github.com/manageiq/manageiq/blob/70edcc40e631a9ef7ed2aaf79d09baf21e4b5726/app/models/mixins/has_network_manager_mixin.rb#L29

It makes sense to set the name only on creation, but if you have a reason otherwise I can change this.

BTW I hope the name has no implication, if you know of one please let me know